src/main/java/org/opentripplanner/middleware/OtpMiddlewareMain.java

```diff
@@ -57,58 +54,37 @@ public class OtpMiddlewareMain {
     public static boolean inTestEnvironment = false;
 
     public static void main(String[] args) throws IOException {
-        // Load configuration.
-        ConfigUtils.loadConfig(args);
-
-        // Initialize template engine
-        TemplateUtils.initialize();
+        CommandLineProcessor processor = new CommandLineProcessor();
+        processor.parseArguments(args);
 
+        // Load configuration.
+        ConfigUtils.loadConfig(processor.getConfigFile());
         // Connect to MongoDB.
         Persistence.initialize();
 
-        MonitoredComponent.initializeMonitoredComponentsFromConfig();
-
-        initializeHttpEndpoints();
+        if (processor.hasEndPoints()) {
+            // This must take place before the main thread is blocked below.
+            TemplateUtils.initialize();
```

Comment:
`TemplateUtils.initialize();` is for notification content... Move this to outside the if block. It is needed for endpoint mode (when saving a trip the first time, monitored trip stuff runs and notifications might be sent then) and in job-only mode. Without that, in jobs-only mode, notifications will not work.